apps/emqx_rule_engine/src/emqx_rule_actions.erl

```diff
@@ -110,8 +112,9 @@ republish(
     Payload = format_msg(PayloadTks, Selected),
     QoS = replace_simple_var(QoSTks, Selected, 0),
     Retain = replace_simple_var(RetainTks, Selected, false),
+    PubProps = maps:get(pub_props, Selected, #{}),
```

Comment:
I am not sure, if `pub_props` always exist on `Selected`, in which case we can pattern-match it in signature above.

Reply:
probably does not exist for messages initiated internally